generator/internal/logging_decorator_rest_generator.cc

```diff
@@ -0,0 +1,170 @@
+    VarsDictionary service_vars,
+    std::map<std::string, VarsDictionary> service_method_vars,
+    google::protobuf::compiler::GeneratorContext* context)
+    : ServiceCodeGenerator("logging_rest_header_path", "logging_rest_cc_path",
+                           service_descriptor, std::move(service_vars),
+                           std::move(service_method_vars), context) {}
+
+Status LoggingDecoratorRestGenerator::GenerateHeader() {
+  HeaderPrint(CopyrightLicenseFileHeader());
+  HeaderPrint(R"""(
+// Generated by the Codegen C++ plugin.
+// If you make any local changes, they will be lost.
+// source: $proto_file_name$
+
+#ifndef $header_include_guard$
+#define $header_include_guard$
+
+)""");
+
+  // includes
+  HeaderLocalIncludes(
+      {vars("stub_rest_header_path"), "google/cloud/internal/rest_context.h",
+       "google/cloud/tracing_options.h", "google/cloud/version.h"});
+  HeaderSystemIncludes(
+      {HasLongrunningMethod() ? "google/longrunning/operations.pb.h" : "",
+       "memory", "set", "string"});
+
+  auto result = HeaderOpenNamespaces(NamespaceType::kInternal);
+  if (!result.ok()) return result;
+
+  // Abstract interface Logging base class
+  HeaderPrint(R"""(
+class $logging_rest_class_name$ : public $stub_rest_class_name$ {
+ public:
+  ~$logging_rest_class_name$() override = default;
+  $logging_rest_class_name$(std::shared_ptr<$stub_rest_class_name$> child,
+                       TracingOptions tracing_options,
+                       std::set<std::string> components);
+)""");
+
+  for (auto const& method : methods()) {
+    if (HasHttpAnnotation(method) && IsNonStreaming(method)) {
+      if (IsResponseTypeEmpty(method)) {
+        HeaderPrintMethod(method, __FILE__, __LINE__, R"""(
+  Status $method_name$(
+      google::cloud::rest_internal::RestContext& rest_context,
+      $request_type$ const& request) override;
+)""");
+      } else {
+        HeaderPrintMethod(method, __FILE__, __LINE__, R"""(
+  StatusOr<$response_type$> $method_name$(
+      google::cloud::rest_internal::RestContext& rest_context,
+      $request_type$ const& request) override;
+)""");
+      }
+    }
+  }
+
+  HeaderPrint(R"""(
+ private:
+  std::shared_ptr<$stub_rest_class_name$> child_;
+  TracingOptions tracing_options_;
+  std::set<std::string> components_;
+};  // $logging_rest_class_name$
+)""");
+
+  HeaderCloseNamespaces();
+  HeaderPrint("\n#endif  // $header_include_guard$\n");
+  return {};
+}
+
+Status LoggingDecoratorRestGenerator::GenerateCc() {
+  CcPrint(CopyrightLicenseFileHeader());
+  CcPrint(R"""(
+// Generated by the Codegen C++ plugin.
+// If you make any local changes, they will be lost.
+// source: $proto_file_name$
+
+)""");
+
+  // includes
+  CcLocalIncludes({vars("logging_rest_header_path"),
+                   "google/cloud/internal/log_wrapper.h",
+                   "google/cloud/status_or.h"});
+  CcSystemIncludes({vars("proto_header_path"), "memory"});
```

Comment:
I think the `#include` for `vars("proto_header_path")` belongs in the generated `.h` file or nowhere.  If you are trying to use IWYU, then it is needed in the header file.  OTOH `vars("stub_rest_header_path")` already provides it, so I am not even sure it is needed there.